lib/utils/logger-utility.js

```diff
@@ -27,24 +27,32 @@ module.exports = (function () {
       for (let item of logBuffer) {
         // display separators for each item before-hand except the first time
         if (item !== logBuffer[0]) {
-          console.warn("----------------------------------------");
+          console.warn("\n----------------------------------------");
         }
         item = JSON.parse(item);
 
         console.warn("[" + item.time + "] - " + LEVEL_MAPPING[item.level] + " - " + item.activity.toUpperCase());
         if (item["request-id"]) {
           console.warn("request-id: " + item["request-id"]);
         }
-        console.warn(item.request.method + " " + item.request.url);
-        console.warn("status code: " + item.response.statusCode + " " + item.response.statusMessage);
+        if (item.request) {
+          console.warn(item.request.method + " " + item.request.url);
+        }
+        if (item.response) {
+          console.warn("status code: " + item.response.statusCode + " " + item.response.statusMessage);
+        }
         if (item.error) {
-          console.warn("error: " + item.error);
+          console.warn("\nerror: " + JSON.stringify(item.error));
+        }
+        if (item.request) {
+          console.warn("\nRequest headers: " + JSON.stringify(item.request.headers));
```

Comment:
it might be better to remove the if so there is no doubt the item.request was not present.
console.warn("\nRequest body: '" + (item.request && item.request.body ? item.request.body : '') + "'.");

Reply:
Agree! updated to consistently show request / response  headers and body...